src/main/java/io/github/fvarrui/javapackager/packagers/Packager.java

```diff
@@ -123,7 +124,9 @@ private void init() throws Exception {
 
 		// sets jdkPath by default if not specified
 		if (jdkPath == null) {
-			jdkPath = new File(System.getProperty("java.home"));
+			TaskJavaUpdater taskJavaUpdater = new TaskJavaUpdater(platform);
+			taskJavaUpdater.execute(jdkVersion, jdkVendor);
+			jdkPath = taskJavaUpdater.jdkPath;
 		}
```

Comment:
Not sure if this is the right place to the the updating/downloading...

Reply:
Also not sure if the default jdkVersion and jdkVendor can be null here.
That should not happen!